DataBiosphere/duos-ui
# DT-944: Fix chairperson edit DAC feature

### Addresses
https://broadworkbench.atlassian.net/browse/DT-944

### Summary
This PR makes a number of changes in function and style. The functional changes are intentional. As a business rule, we allow for Chairs to make changes to their own DACs in terms of both membership and other related details. There are also a number of minor style fixes.

Requires corresponding Consent PR: https://github.com/DataBiosphere/consent/pull/2419

### Testing Notes

1. Update the `env` in your local `config.json` to be `prod` so that you're seeing what we would see in production.
2. Ensure you're testing with a user that does not have Admin permissions. Admins can edit anything so you wouldn't see the current bug in that case.
3. There are actually two versions of this page, one in prod, one in non-prod. To see the same page in non-prod, change your `env` config to `dev` or `staging`.

#### Prod: New screen
![Screenshot 2024-10-30 at 11 41 45 AM](https://github.com/user-attachments/assets/5847eaa4-ab2c-486f-a909-9aec16c3dc8d)

#### Prod: Old screen
![Screenshot 2024-10-30 at 11 43 36 AM](https://github.com/user-attachments/assets/1b998cac-d550-40b8-b0bc-6699e4fb74b6)

#### Non-prod: New screen
![Screenshot 2024-10-30 at 12 23 09 PM](https://github.com/user-attachments/assets/5ff94efa-48e6-459a-9fd5-3bd8de21aae2)

#### Non-prod: Old screen
![Screenshot 2024-10-30 at 12 23 44 PM](https://github.com/user-attachments/assets/9a9848d6-8dd3-478c-a1df-05dbea5b3ad8)

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment
